src/pudl/transform/ferc1.py

```diff
@@ -6509,3 +6540,152 @@ def plants_steam_ferc1(
+    return other_dimensions
+
+
+def table_to_xbrl_factoid_name() -> dict[str, str]:
+    """Build a dictionary of table name (keys) to ``xbrl_factiod`` column name."""
+    return {
+        table_name: transformer().params.xbrl_factoid_name
+        for (table_name, transformer) in FERC1_TFR_CLASSES.items()
+    }
+
+
+@asset(ins={table_name: AssetIn(table_name) for table_name in FERC1_TFR_CLASSES})
+def table_dimensions_ferc1(**kwargs) -> pd.DataFrame:
+    """Build a table of dimensions found in the transformed data tables.
+
+    The dimensions that show up within the transformed tables differ across table and
+    factoid. We need to include the ``xbrl_factoid`` column because these dimensions can
+    differ based on the ``xbrl_factoid``. So we first rename all of the columns which
+    contain the ``xbrl_factoid`` using :func:`table_to_xbrl_factoid_name` rename
+    dictionary.
+
+    Note: It is theoretically possible that we could compile these dimensions based on
+    strictly the parameters (such as using info from ``params.categorize_strings``).
+    Because we did not explictly restrict ourselves to only add categories that are
+    definitely found in the data, we went with using the data tables instead. Also since
+    we added the ``xbrl_factoid`` into the dimensions
+    """
+    table_to_xbrl_factoid_name_dict = table_to_xbrl_factoid_name()
+    tbls = {
+        name: df.assign(table_name=name).rename(
+            columns={table_to_xbrl_factoid_name_dict[name]: "xbrl_factoid"}
+        )
+        for (name, df) in kwargs.items()
+    }
+
+    other_dimensions = get_other_dimensions()
+    dimensions = (
+        pd.concat(tbls.values())[["table_name", "xbrl_factoid"] + other_dimensions]
+        .drop_duplicates()
+        .reset_index(drop=True)
+    )
+    return dimensions
+
+
+@asset(
+    ins={
+        "clean_xbrl_metadata_json": AssetIn("clean_xbrl_metadata_json"),
+        "table_dimensions_ferc1": AssetIn("table_dimensions_ferc1"),
+    }
+)
+def calculation_components_xbrl_ferc1(**kwargs):
+    """Create calculation-compnent table from table-level metadata."""
+    clean_xbrl_metadata_json = kwargs["clean_xbrl_metadata_json"]
+    table_dimensions = kwargs["table_dimensions_ferc1"]
+    # compile all of the calc comp tables.
+    calc_metas = []
+    for table_name, transformer in FERC1_TFR_CLASSES.items():
+        calc_meta = (
+            transformer(xbrl_metadata_json=clean_xbrl_metadata_json[table_name])
+            .process_xbrl_metadata_calculations()
+            .assign(table_name=table_name)
+        )
+        calc_metas.append(calc_meta)
+    # squish all of the calc comp tables then add in the implicit table dimensions
+    calc_components = pd.concat(calc_metas).pipe(
+        make_implict_dimensions_explict_in_calculaiton_components, table_dimensions
+    )
+    return calc_components
+
+
+def make_implict_dimensions_explict_in_calculaiton_components(
```

Comment:
Typo in function name. Should be `_calculation_` not `_calculaiton_`. Maybe just call it `add_explicit_dimensions()` or `make_dimensions_explicit()`?

Reply:
ty for the typo :eyes:. I also like the simplification but i'd like to go with `make_calculation_dimensions_explicit()` because this is very calculation focused.